pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -199,44 +203,49 @@ public synchronized boolean containsFunction(String tenant, String namespace, St
      * @throws IllegalStateException if we are not the leader
      * @throws IllegalArgumentException if the request is out of date.
      */
-    public synchronized void updateFunctionOnLeader(FunctionMetaData functionMetaData, boolean delete)
+    public void updateFunctionOnLeader(FunctionMetaData functionMetaData, boolean delete)
```

Comment:
Shouldn't this be at a higher layer? Like in the rest call handler. For regular commands we check if worker is ready. For this command we shuold check if we are leader and if leaderInit is done. Not sure if metadata mgr should get this logic

Reply:
There are multiple callers of updateFunctionOnLeader.  It made more sense to just wait for the leader to init here